src/HDF5.jl

```diff
@@ -978,19 +994,62 @@ end
 t_commit(parent::Union{HDF5File, HDF5Group}, path::String, dtype::HDF5Datatype) = t_commit(parent, path, dtype, p_create(H5P_LINK_CREATE))
 
 a_create(parent::Union{HDF5File, HDF5Object}, name::String, dtype::HDF5Datatype, dspace::HDF5Dataspace) = HDF5Attribute(h5a_create(checkvalid(parent).id, name, dtype.id, dspace.id), file(parent))
+
+function _prop_set!(p::HDF5Properties, name::Symbol, val, check::Bool = true)
```

Comment:
Is this `check` arg really needed? I'm not sure, but it seems a bit superfluous?   I think the rationale is that we can squeeze a bit of perf without having to  `error` check, but perhaps the compiler is good enough that we don't need such fine grained control?

Reply:
EDIT:

Actually NVM I don't think it's that feasible given the way the code is structured as is, and would remove some important error checking.

Reply:
> Is this check arg really needed, It seems a bit superfluous?

The reason I added it was to maintain the current behavior — `p_create` does not error even if you pass it properties that are not appropriate for a particular property class, while `setindex!` did (by way of just throwing when the underlying `h5p_set_*` function errored). The two functions now set `false` and `true`, respectively, to maintain that.